packages/mobx-state-tree/__tests__/core/identifier.test.ts

```diff
@@ -268,3 +268,44 @@ test("identifierAttribute of the type", () => {
     const M3 = types.model({ myId: types.optional(types.identifier, () => "hi") })
     expect(M3.identifierAttribute).toBe("myId")
 })
+
+test("items detached from arrays don't corrupt identifierCache", () => {
```

Comment:
**praise:** @scytacki - thank you for the detailed test cases here. Really make it clear what's goin' on.